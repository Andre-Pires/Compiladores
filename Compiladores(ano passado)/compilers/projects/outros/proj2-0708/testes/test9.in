int sum;

typedef struct {
  int x;
  int y;
} Coord;

struct Coord* A;
struct Coord* B;

void init(){
  sum = 0;
  A = nil;
  B = nil;
}

void f(int n){
  int m;
  m=n;
  while(m>0){
    sum = sum + m;
  }
}

