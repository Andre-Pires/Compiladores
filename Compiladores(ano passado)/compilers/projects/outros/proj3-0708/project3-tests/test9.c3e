init: 	n = call iread, 0
	sum = 0
	i = 0
L1:	t1 = i
	t2 = n
	if t1 < t2 goto L2
	goto L3
L2:	t3 = call iread, 0
	t4 = i
	xy[t4] = t3
	t5 = i
	t6 = xy[t5]
	t7 = sum + t6
	sum = t7
	t8 = i + 1
	i = t8
	goto L1
L3:	t9 = sum
	putparam t9
	call iwrite, 1
	halt
