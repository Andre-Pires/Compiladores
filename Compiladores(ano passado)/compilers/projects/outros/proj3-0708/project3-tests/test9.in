/* straight line code with scalars, arrays one input and while loop */
int xy[10];
void main(){
  int n;
  int i;
  int sum;

  n = iread();
  sum = 0;
  i = 0;
  while (i < n){
   xy[i] = iread();
   sum = sum + xy[i];
   i = i + 1;
  }
  iwrite(sum);
}
