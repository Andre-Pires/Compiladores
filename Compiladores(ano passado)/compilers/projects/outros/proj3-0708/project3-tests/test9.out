C3E Simulator 07/10/07, 22:55:13
17
*** Symbol table dump ***
init = 0
n = 5
sum = 17
i = 5
L1 = 3
t1 = 5
t2 = 5
L2 = 7
L3 = 17
t3 = 7
t4 = 4
xy = 0
t5 = 4
t6 = 7
t7 = 17
t8 = 5
t9 = 17
