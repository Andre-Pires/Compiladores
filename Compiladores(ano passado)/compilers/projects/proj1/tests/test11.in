sizeof
#include "test9.in"
else
