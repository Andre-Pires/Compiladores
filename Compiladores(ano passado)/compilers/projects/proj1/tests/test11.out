1 259 sizeof
1 257 qwerty
1 258 13456
1 271 int
1 257 x
1 59 ;
4 271 int
4 257 s
4 40 (
4 271 int
4 257 x
4 41 )
4 123 {
5 278 return
5 257 x
5 43 +
5 258 1
5 59 ;
5 125 }
3 277 while
3 278 return
3 276 else
#Total Number of Tokens Matched was 22
